### PR TITLE
docs: k8s deployment docs [DET-9449]

### PR DESCRIPTION
**Pull request by eecsliu**
## Description
Update kubernetes custom master deployment docs. They haven't been updated since we switched to using devcluster.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-k8s/k8s-dev-guide.rst [hunk 1]
```diff
@@ -95,57 +94,79 @@ In addition to installing these packages, ``.bashrc`` needs to be updated with n
...
-   def main() -> None:
-     ...
-     try: # comment out all lines in here except for these four:
-       master = run_master()
-       ...
-       master.start()
-       wait_for_server(8080)
-       ...
-       master.join()
+Once the dependencies are installed, you will need to prepare the repository to run `devcluster`, a
```
Comment (dzhu) [inline docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-k8s/k8s-dev-guide.rst:114]:
```suggestion
Once the dependencies are installed, you will need to prepare the repository to run ``devcluster``, a
```

File: docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-k8s/k8s-dev-guide.rst [hunk 2]
```diff
@@ -95,57 +94,79 @@ In addition to installing these packages, ``.bashrc`` needs to be updated with n
...
+tool for running Determined. First, enter the Determined reposiroty and run:
```
Comment (dzhu) [inline docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-k8s/k8s-dev-guide.rst:115]:
```suggestion
tool for running Determined. First, enter the Determined repository and run:
```

File: docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-k8s/k8s-dev-guide.rst [hunk 3]
```diff
@@ -95,57 +94,79 @@ In addition to installing these packages, ``.bashrc`` needs to be updated with n
...
+            type: "kubernetes"
+            namespace: default
+            max_slots_per_pod: 1
+            master_service_name: <name of determined master service from `kubectl get services`>
+
+         log:
+            level: debug
+         root: tools/build
+
+We are now ready to build and run the Determined Master! From the Determined repo, run ``devcluster
```
Comment (dzhu) [inline docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-k8s/k8s-dev-guide.rst:168]:
```suggestion
We are now ready to build and run the Determined master! From the Determined repo, run ``devcluster
```

File: docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-k8s/k8s-dev-guide.rst [hunk 4]
```diff
@@ -95,57 +94,79 @@ In addition to installing these packages, ``.bashrc`` needs to be updated with n
...
+--no-guess-host`` to build and run the Master.
```
Comment (dzhu) [inline docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-k8s/k8s-dev-guide.rst:169]:
```suggestion
--no-guess-host`` to build and run the master.
```

File: docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-k8s/k8s-dev-guide.rst [hunk 5]
```diff
@@ -95,57 +94,79 @@ In addition to installing these packages, ``.bashrc`` needs to be updated with n
...
-We are now ready to build and run the Determined Master! From the Determined repo, run ``make all``
-to build and ``make -C tools run`` to start the Master.
+   startup_input: "p"
+
+   cwd: /root/determined
+
+   commands:
+   p: make -C harness build  # rebuild Python
+   w: make -C webui build    # rebuild Webui
+   c: make -C docs build     # rebuild doCs
```
Comment (dzhu) [inline docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-k8s/k8s-dev-guide.rst:133]:
```suggestion
   c: make -C docs build     # rebuild docs
```

File: docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-k8s/k8s-dev-guide.rst [hunk 6]
```diff
@@ -95,57 +94,79 @@ In addition to installing these packages, ``.bashrc`` needs to be updated with n
...
 
-We are now ready to build and run the Determined Master! From the Determined repo, run ``make all``
-to build and ``make -C tools run`` to start the Master.
+   startup_input: "p"
+
+   cwd: /root/determined
+
+   commands:
+   p: make -C harness build  # rebuild Python
+   w: make -C webui build    # rebuild Webui
```
Comment (dzhu) [inline docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-k8s/k8s-dev-guide.rst:132]:
```suggestion
   w: make -C webui build    # rebuild WebUI
```

**Review by dzhu — APPROVED** [review]
Gave it a quick editing pass; I haven't gone through the instructions in detail, but it's surely better than what was here before.

**Review by tara-det-ai — APPROVED** [review]
LGTM

**Pull request merged**